File: backend/crates/akashic-extraction/tests/fixtures/csharp/02-interface.cs
namespace MyApp.Contracts
{
    /// <summary>Defines a shape with area computation.</summary>
    public interface IShape
    {
        double Area();
        double Perimeter();
        string Describe();
    }

    /// <summary>An interface for serializable objects.</summary>
    public interface ISerializable
    {
        string Serialize();
    }
}

File: backend/crates/akashic-extraction/tests/fixtures/php/02-interface.php
<?php

namespace App\Contracts {
    interface Shape
    {
        public function area(): float;

        public function name(): string;
    }

    trait Describable
    {
        public function describe(): string
        {
            return $this->name();
        }

        protected function tag(): string
        {
            return "shape";
        }
    }
}
